crates/crypto_api/src/crypto_system/crypto_system_test.rs

```diff
@@ -251,6 +278,74 @@ mod test {
             Ok(())
         }
 
+        fn hash_sha256_bytes(&self) -> usize {
+            32
+        }
+        fn hash_sha512_bytes(&self) -> usize {
+            64
+        }
+        fn pwhash_salt_bytes(&self) -> usize {
+            8
+        }
+        fn pwhash_bytes(&self) -> usize {
+            16
+        }
+
+        fn hash_sha256(
+            &self,
+            hash: &mut Box<dyn Buffer>,
+            data: &Box<dyn Buffer>,
+        ) -> CryptoResult<()> {
+            if hash.len() != self.hash_sha256_bytes() {
+                return Err(CryptoError::BadHashSize);
+            }
+
+            let mut hasher = sha2::Sha256::new();
+            hasher.input(data.read_lock().deref());
+            hash.write(0, &hasher.result())?;
```

Comment:
Just want to make sure you made a conscious decision to unwrap here as opposed to returning the error back in the `CryptoResult`. Same with the other implemented hash functions. I don't know why the write here would even fail- perhaps you already checked for it by verifying the hash lengths are correct?

Reply:
@struktured - I don't see an unwrap here. The `?` will forward any write errors up to the caller.